modules/standard/Socket.md

```diff
@@ -0,0 +1,235 @@
+- Calling in `listen` without `reuseaddr` can return an error that some other socket is already bound to the specified address. This can also be a result of re-running the program after only a short waiting period where the `bind` may find address as already in use.
+
+The `tcpListener` instance will have methods on it to call
+- `accept`: method will return a `tcpConn` instance of any new incoming connection otherwise will yield if no new connection is available at the instant. The communication can then proceed through the `tcpConn` instance returned
+  ```python
+  proc tcpListener.accept(in timeout:timeval = new timeval(-1,0)) : tcpConn
+  ```
+  `accept` will throw`TimeoutError` on not being able to accept any new connection
+  before `timeout`.
+- `close`,
+  calling close on `tcpListener` will close the socket server and it won't accept anymore new connections.
+  ```python
+  proc tcpListener.close()
+  ```
+- `family` returns the family of socket `AF_INET` or `AF_INET6`
+- `addr` will return an `ipAddr` instance providing details about the address and port at which socket is listening.
+  ```python
+  proc tcpListener.addr : ipAddr
+  ```
+
+The additional support with `tcpConn` and `tcpListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+  This method will set/unset TCP_NODELAY for provided socket.
+  ```python
+  proc nagle(socket:tcpConn, tcpNodeDelay:bool = 1)
+  proc nagle(socket:tcpListener, tcpNodeDelay:bool = 1);
+  proc nagle(socket:udpSocket, tcpNodeDelay:bool = 1);
+  ```
+- `quickAck`
+  This method will set/unset TCP_QUICKACK for provided socket.
+  ```python
+  proc quickAck(socket:tcpConn, tcpQuickAck:bool = 1)
+  proc quickAck(socket:tcpListener, tcpQuickAck:bool = 1);
+  proc quickAck(socket:udpSocket, tcpQuickAck:bool = 1);
+  ```
+
+## UDP Support
+
+Support for UDP protocol will be through a separate constructor for `udpSocket`. They need not be binded or connected as they are connectionless socket.
+```python
+const socketServ:udpSocket = new udpSocket(family=IPFamily.IPv4)
+```
+
+A generic `bind` function will be used which will take in any of the socket three object and bind it to provided `host` , `port`.
+
+**Method**
+
+```
+proc bind(socket, in address:ipAddr, reuseAddr = true)
+```
+
+**Parameters**
+
+```
+socket : tcpListener|tcpConn|udpSocket - socket object to be bound
+address: contains address info about where to bind the socket.
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+Bind is responsible for binding any socket to a specified address and port. Here the socket can be anything either a UDP or a TCP Socket. listen under the hood will can in bind to bind `tcpListener` to address and port. With either tcpConn or udpSocket that isn't a requirement as the system assigns any free port and localhost address to them by default.
```

Comment:
I'm not sure I know what "will can in bind to bind" was supposed to mean, sorry :(

Reply:
*call 😅  will make the change